File: array/1018_binary_prefix_divisible_by_5/1018_binary_prefix_divisible_by_5.go
package main

import "fmt"

func prefixesDivBy5(A []int) []bool {
	num, result := 0, make([]bool, len(A))
	for k, v := range A {
		num = num<<1 + v
		if num%5 == 0 {
			result[k] = true
		}
		num %= 10

	}
	return result
}

func main() {
	A := []int{0, 1, 1}
	res := prefixesDivBy5(A)
	fmt.Println(res)
}

File: array/1018_binary_prefix_divisible_by_5/1018_binary_prefix_divisible_by_5.py
from typing import List

class Solution:
    def prefixesDivBy5(self, A: List[int]) -> List[bool]:
        result = []
        for i in range(len(A)):
            if i: 
                A[i] = A[i] + (A[i-1] << 1)
            if A[i] % 5 == 0: 
                result.append(True)
            else:
                result.append(False)
        return result  



A = [0,1,1]
res = Solution().prefixesDivBy5(A)
print(res)